appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/AssetList.java

```diff
@@ -110,16 +110,15 @@ private void refreshAssetList() {
               nodeName.length());
 
         String fileSuffix = node.getProjectId() + "/" + node.getFileId();
-        String treeItemText = "<span style='cursor: pointer'>" + nodeName + "</span>";
+        String treeItemText = "<span style='cursor: pointer'>";
         if (StorageUtil.isImageFile(fileSuffix)) {
-          Image mediaIcon = new Image(images.mediaIconImg());
-          treeItemText = "<span>" + mediaIcon + nodeName + "</span>";
+          treeItemText += new Image(images.mediaIconImg()) + nodeName + "</span>";
         } else if (StorageUtil.isAudioFile(fileSuffix )) {
-          Image mediaIcon = new Image(images.mediaIconAudio());
-          treeItemText = "<span>" + mediaIcon + nodeName + "</span>";
+          treeItemText += new Image(images.mediaIconAudio()) + nodeName + "</span>";
+        } else if (StorageUtil.isVideoFile(fileSuffix )) {
+          treeItemText += new Image(images.mediaIconVideo()) + nodeName + "</span>";
         } else {
-          Image mediaIcon = new Image(images.mediaIconVideo());
-          treeItemText = "<span>" + mediaIcon + nodeName + "</span>";
+          treeItemText += nodeName + "</span>";
```

Comment:
You didn't need to move this inside of each branch, you could have left it where it was in the previous change. `nodeName` and `</span>` are always added, so you don't need to duplicate it over every branch.

Reply:
Haha, how could I be so illogical……